src/microbe_stage/CompoundBag.cs

```diff
@@ -13,20 +13,23 @@ public class CompoundBag : ICompoundStorage
     private readonly HashSet<Compound> usefulCompounds = new();
 
     /// <summary>
-    ///   Creates a new bag
+    ///   This costructor is used for loading saves
     /// </summary>
-    /// <param name="capacity">Specifies the initial capacity of the compound bag</param>
+    public CompoundBag(Dictionary<Compound, float> compoundCapacities)
+    {
+        CompoundCapacities = compoundCapacities;
+    }
+
     public CompoundBag(float capacity)
     {
-        Capacity = capacity;
+        SetCapacityForAllCompounds(capacity);
     }
 
     /// <summary>
     ///   How much of each compound this bag can store.
-    ///   Currently a CompoundBag can hold the same amount of each compound.
     /// </summary>
     [JsonProperty]
-    public float Capacity { get; set; }
+    public Dictionary<Compound, float> CompoundCapacities { get; private set; } = new();
```

Comment:
The internal details of how the capacity is stored should not be leaked. This should be made private. I created the `GetCapacityForCompound` method for a reason. And second problem is that if a mod for example adds more compounds or we add more compounds in a later update, this is so going to blow up and cause a game crash somewhere.

So here's my suggestion have a "nominal capacity" private field and a nullable dictionary of specific capacities also a private field. Then modify the GetCapacityForCompound to first check if a specific capacity exists for a compound, and if it does return it, otherwise return the nominal capacity. I think with those changes we'd have a design that I can think of being good enough.

Reply:
Ok, on it